### ISSUE TITLE
[info] How to use TsInputComponent Custom Error Validation with "ngModel"

### ISSUE THREAD

**Issue by divyameher** [GetTerminus/terminus-ui#702]
Are we restricting `TsInputComponent ` Custom Error Validation only to `formControl`?

In the component code of `TsInputComponent` I'm seeing below code snippet with `*ngIf="formControl"` :
```
 <ts-validation-messages
    class="qa-input-validation-messages"
    *ngIf="formControl"
    [control]="formControl"
    [validateOnChange]="validateOnChange"
  ></ts-validation-messages>
```

And Also observed that UX of `Validation Messages` in Input is not consistent for `required` validation and other `custom validations`
This can be observed in the [example](https://stackblitz.com/edit/terminus-ui-demos-kwbvjj?file=app%2Fcomponents%2Finput%2Finput.component.html) for Form between `My Input` and `My 2nd Input` under `Input` Demo

**Comment by benjamincharity** [comment]
Hello! Thanks for taking the time to create an issue around this. 🙏 

You have definitely found an area where we skimped due to time constraints. 😞 
 Internally we are using `ReactiveForms` everywhere so we punted on supporting `ngModel`. Currently we have a very small team working on this library while several larger teams are trying to consume the library, so we've purposefully chosen the minimum viable path (for example: our custom validators do not have a `directive` version so they can currently only be used on reactive forms).

That being said, this is definitely something that should/will be supported. So let me make sure I fully understand everything you are addressing:


### a) Validation messages do not work with `ngModel`

Correct. This shouldn't be terrible difficult since `ngModel` has validator helpers built in. As mentioned above, our custom validators don't have a directive implementation so we would need to add those also. I will create an issue and link back here.

### b) UX of Validation Messages in Input is not consistent for required validation and other custom validations

b.1) If I understand correctly, this is referring to the fact that if we set the `isRequired` flag on an input we see the required asterisk `*`, but if the simply add the required validation to the reactive form there is no asterisk added.

This definitely should be fixed. I think it should be fairly easy.. Rather than use the `isRequired` property directly in the template, we can create a getter that returns true/false based on that flag _and_ any existing required validator.

b.2) Can you explain a bit more about the 'other `custom validations`' part you mentioned?

**Comment by divyameher** [comment]
@benjamincharity  As you can see in this [example](https://stackblitz.com/edit/terminus-ui-demos-kwbvjj?file=app%2Fcomponents%2Finput%2Finput.component.html) of Input Demo, `mat-input-underline` is taking red color for `required` and default color for `My 2nd Input`
![image](https://user-images.githubusercontent.com/24610542/36597572-8f0ade92-18cf-11e8-8702-f8bfe5eadf13.png)

**Comment by benjamincharity** [comment]
Ahh yes I see. Thanks for the clarification! I believe that will be fixed with #705

**Comment by benjamincharity** [comment]
Closing this in favor of https://github.com/GetTerminus/terminus-ui/issues/703
